### PR TITLE
Ignore last broken line in FileDb

### PR DESCRIPTION
**Pull request by deffrian**
Fixes Closes Resolves #
#5700 
During file update client might be terminated, so last line might be broken.

## Changes

- Do not throw on malformed data

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by benaadams — APPROVED** [review]
Does the file get overwritten when broken so its repaired for next time?

**Comment by LukaszRozmej** [comment]
Maybe a better idea is to also buffer a write so it happens completely or not at all: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Db/SimpleFilePublicKeyDb.cs#L128-L134

**Comment by deffrian** [comment]
@benaadams Yes, it overwrites the file with latest state.

**Comment by benaadams** [comment]
Just had this issue so can confirm it fixes

image

File: src/Nethermind/Nethermind.Db/SimpleFilePublicKeyDb.cs [hunk 1]
```diff
@@ -122,15 +122,21 @@ private void CommitBatch()
...
+                using StreamWriter fileWriter = new(DbPath);
                 foreach ((byte[] key, byte[] value) in snapshot)
                 {
                     if (value is not null)
                     {
-                        key.StreamHex(streamWriter);
-                        streamWriter.Write(',');
-                        value.StreamHex(streamWriter);
-                        streamWriter.WriteLine();
+                        StringBuilder lineBuilder = new();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Db/SimpleFilePublicKeyDb.cs:130]: very bad performance, to fix:
1. Reuse StringBuilder.
2. Write with:

`foreach (c in stringBuilder.GetChunks()) c.Span`

I expect this to have only one span almost always as it is small

3. Maybe initialize StringBuilder with some decently high capacity so it won't be chunked

**Review by LukaszRozmej — APPROVED** [review]

**Review by benaadams — APPROVED** [review]

**Pull request merged**